### PR TITLE
feat: add git check

### PR DESCRIPTION
**Pull request by Lampese**

### CODE REVIEW

**Comment by Lampese** [comment]
Test CI seems doesn't work, is it my fault?

**Comment by bzy-debug** [comment]
> Test CI seems doesn't work, is it my fault?

Might be related to secrets restriction in forked PR. I'm looking into it.

File: Cargo.toml [hunk 1]
```diff
@@ -90,6 +90,7 @@ thiserror = "1.0"
 test-log = "0.2"
 fs4 = { version = "0.8.3", features = ["sync"] }
 ariadne = { version = "0.4.1", features = ["auto-color"] }
+git2 = "0.19.0"
```
Comment (lijunchen) [inline Cargo.toml]: This causes the binary size of `moon` to increase from 3.5M to 6.4M (tested on macOS), and binary size is one of our optimization goals.

Comment (Lampese) [reply Cargo.toml]: Ok, then I'll try to split a part of git2 to here

**Comment by Lampese** [comment]
Test CI seems doesn't work again? @bzy-debug

File: crates/moonutil/src/git.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        .stderr(std::process::Stdio::null())
+        .current_dir(path)
+        .status();
+    match output {
+        Ok(out) => out.success(),
+        _ => false,
+    }
+}
+
+pub fn create_git_repo(path: &Path) {
```
Comment (Young-Flash) [inline crates/moonutil/src/git.rs]: a better name would be `git_init_repo`

File: crates/moon/tests/test_cases/mod.rs [hunk 3]
```diff
@@ -489,7 +489,6 @@ fn test_moon_new_exist() {
     );
 
     assert!(res.contains("Created hello"));
-    assert!(res.contains("Initialized empty Git repository"));
```
Comment (Young-Flash) [inline crates/moon/tests/test_cases/mod.rs]: IMO this could be retained to verify that `git init` have been executeed

**Review by Young-Flash — APPROVED** [review]
Mostly LGTM, thanks

**Comment by Lampese** [comment]
IMO this coverage check action may be some wrong?

**Pull request merged**